lib/cloud/file-storage/gridfsStorage.js

```diff
@@ -0,0 +1,73 @@
+var q = require('q');
+var fs = require('fs');
+var mongo = require('mongodb');
+var Grid = require('gridfs-stream');
+var MongoClient = require('mongodb').MongoClient
+
+var gfs;
+
+function init(storageConfiguration) {
+  var config = storageConfiguration;
+  if (!config || !config.mongoUrl) {
+    throw Error("Missing mongoUrl parameter for GridFs storage");
+  }
+  MongoClient.connect(config.mongoUrl, function(err, connection) {
+    if (err) {
+      return console.log("Cannot connect to mongodb server. Gridfs storage will be disabled");
+    }
+    console.log("Connected successfully to storage server");
+    gfs = Grid(connection, mongo);
+  });
+}
+
+function writeFile(namespace, fileName, fileLocation) {
+  if (!gfs) {
```

Comment:
I think `gfs` should be a promise to take account of the async mongo connection.

Then you could have `gfs.then(...)`